src/app/clusters/test-cluster-server/test-cluster-server.cpp

```diff
@@ -137,6 +145,36 @@ SimpleEnum gSimpleEnums[kAttributeListLength];
 size_t gSimpleEnumCount = 0;
 Structs::NullablesAndOptionalsStruct::Type gNullablesAndOptionalsStruct;
 
+void emberAfUnitTestingClusterTestBatchWork(AsyncBatchCommandsWorkData * asyncWorkData)
+{
+    auto commandHandleRef = std::move(asyncWorkData->asyncCommandHandle);
+    auto commandHandle    = commandHandleRef.Get();
+    if (commandHandle == nullptr)
+    {
+        // Very weird that we are even in here, what set this to nullptr?
+        Platform::Delete(asyncWorkData);
+        return;
+    }
+
+    uint8_t buffer[800];
```

Comment:
Suggest ScopedMemoryBuffer here to reduce stack usage.

Reply:
Since this is called in a timer callback I figured there would be enough space on the stack for 800. ScopedMemoryBuffer has a chance to fail allow if heap is low which is why I went with putting this on the stack.

I can use ScopedMemoryBuffer is you actually think there is a risk of stackoverflow, or if there is some other reason it shouldn't be on the stack. I just don't see the value of moving this off the stack